sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2147,6 +2147,12 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_ADD_DIRECTORY_USING_RECURSIVE = buildConf("spark.sql.legacy.addDirectory.recursive")
```

Comment:
spark.sql.legacy.addDirectory.recursive.enabled?

Reply:
Looks ok and I'll do followup.

Reply:
https://github.com/apache/spark/pull/27372/files

Reply:
Hey guys, I think the config is not consistent between name and default value.

If we want it to be a real legacy config, then, I think its name should be like `spark.sql.legacy.addSingleFileInAddFile` and default to `false`. If not, then, we may name it as `spark.sql.allowAddDirectoryInAddFile` and default to `true`.

Any ideas? @maropu @iRakson 

BTW, I'm going to revise a batch of boolean configs in order to comply with our [new config name policy](http://apache-spark-developers-list.1001551.n3.nabble.com/DISCUSS-naming-policy-of-Spark-configs-td28875.html). So, I'll address the consensus here.

Reply:
Yeah, please go ahead.

Reply:
+1

Reply:
So, which one do you prefer? Legacy or not legacy? Personally, I prefer `not legacy` for this new feature.

Reply:
I'm going to pick `spark.sql.allowAddDirectoryInAddFile`. Thanks!

Reply:
We need to keep the old behaviour? I personally think the legacy one looks ok to me.

Reply:
yup, +1 for @maropu's

Reply:
I guess, it should be `spark.sql.allowAddDirectoryInAddFile` and default set to `true`. If default is set to false then we will go back to our previous behaviour which do not allow adding directory using `ADD FILE`.

Reply:
Sorry guys, forget to tell you that this one wasn't included in that renaming PR. Because the change we need here doesn't related to the naming policy. So, I think we need to separate follow up for it.

@iRakson Can you send a follow up to rename it according to your own thoughts?

Reply:
I will raise a followup for this one.